build.sh

```diff
@@ -36,35 +42,47 @@ bash ./start-rc.sh
 sleep 10
 
 # Run tests (JUnit plugin) with race detection
-for pkg in $(go list $CLIENT_IMPORT_PATH/...);
-do
-    if [[ $pkg != *"vendor"* ]]; then
-      echo "testing with race detection on ... $pkg"
-      go test -race -v  $pkg
-    fi
-done
+#for pkg in $(go list $CLIENT_IMPORT_PATH/...);
+#do
+#    if [[ $pkg != *"vendor"* ]]; then
+#      echo "testing with race detection on ... $pkg"
+#      go test -race -v  $pkg
+#
+#    fi
+#done
+
+# Run vet tools (Compiler warning plugin)
+go vet $CLIENT_IMPORT_PATH > vet.txt
+
+# if --with-coverage flag is not given then dont generate coverage report.
+#if [[ $* != *--with-coverage* ]] ;then
+#    exit 0;
+#fi
 
 
+go get github.com/t-yuki/gocover-cobertura
+go get github.com/tebeka/go2xunit
 # Run tests (JUnit plugin)
-echo "mode: set" > coverage.out
+
+echo "mode: atomic" > coverage.out
+
 for pkg in $(go list $CLIENT_IMPORT_PATH/...);
 do
     if [[ $pkg != *"vendor"* ]]; then
       echo "testing... $pkg"
-      go test -v -coverprofile=tmp.out -coverpkg ${PACKAGE_LIST} $pkg >> test.out
+      go test -race -covermode=atomic  -v -coverprofile=tmp.out ${pkg}
```

Comment:
We don't need atomic run count. `covermode` `set` is enough.

Reply:
When running race and coverage together the most secure way is to do it with atomic run count otherwise when doing coverage race detector gives race condition since the extra lines that are added for coverage are not atomic .